CU-29aq37e Populate data (application term, position, and subteam) taken from th…

IvanYu327 opened the pull request:
…e application instance

### [CLICKUP TICKET LINK](https://app.clickup.com/t/r17re8)

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fill email template with data from emailData to populate application term and the subteam and position the applicant is applying for instead of using mockData

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)

IvanYu327 committed:
Populate data (application term, position, and subteam) taken from the application instance

jennchenn commented:
Task linked: [CU-29aq37e Populate data taken from the application instance](https://app.clickup.com/t/29aq37e)

jmawloo commented on src/frontend/pages/recruitment/DecisionPage/DecisionPage.js:
```diff
@@ -166,9 +187,9 @@ const DecisionPage = () => {
-    nextTerm: 'Spring 2022',
+    // position: 'Fullstack Developer',
+    // subteam: 'Web',
+    // nextTerm: 'Spring 2022',
```
Don't need these lines now, can remove them :)

jmawloo commented on src/frontend/pages/recruitment/DecisionPage/DecisionPage.js:
```diff
@@ -151,6 +151,27 @@ const DecisionPage = () => {
   ];
 
+  const formatTerm = (currentTerm) => {
+    const term = currentTerm?.slice(0,1)+currentTerm?.slice(1,-5).toLowerCase();
```
From my understanding currentTerm is represented as TERM-YEAR, so would be better if we could split the string by the '-' delimiter and then capitalize in order to account for all string lengths for TERM.

Also would be awesome if you could add move this function to utils.js :)

jmawloo reviewed (changes requested):
Nice work, just a few comments :)

IvanYu327 committed:
Update DecisionPage.js

removed comments

IvanYu327 committed:
removed comments
made requested changes to formatting method
moving the formatTerm method to utils.js

IvanYu327 committed:
Merge branch 'CU-29aq37e_Populate-data-taken-from-the-application-instance_Ivan-Yu' of https://github.com/waterloop/executive-dashboard into CU-29aq37e_Populate-data-taken-from-the-application-instance_Ivan-Yu
